### ISSUE TITLE
Questions on reproducing training from scratch (77% in Table 1)

### ISSUE THREAD

**Issue by June01** [TengdaHan/CoCLR#30]
Hi Tengda, thanks for these detailed answers. I looked into all of them, seems no detailed training instruction is given on using main_classifier.py to train from scratch. The thing is, I train on UCF101 with rgb from scratch, after 500 epochs, the reported validation accuracy is 46.1%, while in test set it is only 3.41% (center crop only, top1). The detailed commands are as below:

-- Training
```
CUDA_VISIBLE_DEVICES=0,1 python main_classifier.py  --train_what all --epoch 500 --batch_size 24 --lr 1e-3 --wd 1e-3 --dropout 0.9 --schedule [60, 80]
```

-- Testing
```
CUDA_VISIBLE_DEVICES=0,1 python main_classifier.py --test epochxxx.pth --ten_crop
```
Would you like to have a quick look and help me to figure it out which configuration I made wrong? Though my computation resources is not enough, it is hard to understand why there is such a big gap between validation accuracy and test accuracy? My sincere appreciation.

**Comment by TengdaHan** [comment]
Try command like:
```sh
CUDA_VISIBLE_DEVICES=0,1 python main_classifier.py  --train_what all --epoch 500 --batch_size 24 --lr 1e-3 --wd 1e-3 --dropout 0.9 --schedule [300, 400]
```
For step-wise lr schedule, reduce the learning rate too early is not good.

Test set performance looks like a random guess, are you sure the model was successfully loaded?
If you want to figure out whether the code has problem, try to test this finetune-ed checkpoint (with the linear classifier layer): http://www.robots.ox.ac.uk/~htd/coclr/Finetune-InfoNCE-ucf101-128-s3d-ep324.pth.tar

I got the logfile like:
```
0-00:08:49 CenterCrop:
0-00:08:49 test Epoch: [324] Mean: Acc@1: 0.7687 Acc@5: 0.9347
Last: Acc@1: 0.7687 Acc@5: 0.9347
0-00:44:08 FiveCrop:
0-00:44:08 test Epoch: [324] Mean: Acc@1: 0.7819 Acc@5: 0.9416
Last: Acc@1: 0.7819 Acc@5: 0.9416
0-01:31:11 TenCrop:
0-01:31:11 test Epoch: [324] Mean: Acc@1: 0.7838 Acc@5: 0.9440
Last: Acc@1: 0.7838 Acc@5: 0.9440
```

**Comment by June01** [comment]
Cool! Mine is like:

```
center-crop result:
Mean: Acc@1: 0.0093 Acc@5: 0.0636
```

So I guess you know what's wrong?

**Comment by youcaiSUN** [comment]
Hi tengda,

Thank for your kind sharing. 
I train s3d from scratch using the above command you provided.
Note that the rgb lmdb is constructed from the raw extracted video frame of ucf101 (do not resized to 256). The size of my custom lmdb is ~9GB, which is much smaller than yours (~28GB). 

The best validation accuracy is  Acc@1: 0.6277 Acc@5: 0.8569 (epoch=468), and the test accuracy is:
0-00:09:48	center-crop:
0-00:09:48	test Epoch: [468]	Mean: Acc@1: 0.7061 Acc@5: 0.9231
0-00:52:31	five-crop:
0-00:52:31	test Epoch: [468]	Mean: Acc@1: 0.7129 Acc@5: 0.9299
0-01:49:16	ten-crop:
0-01:49:16	test Epoch: [468]	Mean: Acc@1: **0.7166** Acc@5: 0.9302

Since there is a 7% gap, will the custom lmdb lead to this?

**Comment by June01** [comment]
Hi @youcaiSUN, as I have been able to reproduce 77%, I think it is better to understand what's going on if you make it clear how much different your customized lmdb with Tengda's.

**Comment by youcaiSUN** [comment]
@June01 Thanks very much!

**Comment by June01** [comment]
> @June01 Thanks very much! Do your use the above command provided by the author to reproduce 77%?

Not exactly. I use two gpus with batch size=96, so that I am able to reproduce 77% at epoch 500. Not sure how much difference it will make by different batch size.

**Comment by youcaiSUN** [comment]
Is 96 the total batch size or that on a single gpu?

**Comment by June01** [comment]
> Is 96 the total batch size or that on a single gpu?

Single. So it is 96x2 in total.
